s4e-web/src/app/views/settings/components/generic-list-view/generic-list-view.component.spec.ts

```diff
@@ -0,0 +1,133 @@
+import {async, ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {GenericListViewComponent} from './generic-list-view.component';
+import {GenericListViewModule} from './generic-list-view.module';
+import * as Factory from 'factory.ts';
+import {take} from 'rxjs/operators';
+import {Component, DebugElement} from '@angular/core';
+import {By} from '@angular/platform-browser';
+
+interface Item {
+  id: number;
+  name: string;
+}
+
+const CustomItemFactory = Factory.makeFactory<Item>({
+  id: Factory.each(i => i),
+  name: Factory.each(i => `Sample Item #${i % 10}`)
+});
+
+describe('GenericListViewComponent', () => {
+  let component: GenericListViewComponent;
+  let fixture: ComponentFixture<GenericListViewComponent>;
+  let de: DebugElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [GenericListViewModule]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(GenericListViewComponent);
+    component = fixture.componentInstance;
+    de = fixture.debugElement;
+    component.loading = false;
+    component._items = CustomItemFactory.buildList(10);
+  });
+
+  describe('searchable', () => {
+    beforeEach(() => {
+      component.searchable = true;
+      fixture.detectChanges();
+    });
+
+    it('should search offline', fakeAsync(() => {
+      component.queryFc.setValue('#3');
+      tick(250);
+      expect(component.items.length).toBe(1);
+    }));
+
+    it('should search callback', async () => {
+      component.offlineSearch = false;
+      component.queryFc.setValue('#3');
+      expect(await component.onquery.pipe(take(1)).toPromise()).toEqual('#3');
+    });
+  });
+
+  describe('non searchable', () => {
+    beforeEach(() => {
+      fixture.detectChanges();
+    });
+
+    it('should create', () => {
+      expect(component).toBeTruthy();
+    });
+
+    it('should show loading', () => {
+      expect(de.query(By.css('[data-test-loading=""]'))).toBeFalsy();
+      component.loading = true;
+      fixture.detectChanges();
+      expect(de.query(By.css('[data-test-loading=""]'))).toBeTruthy();
```

Comment:
To be consistent, what naming type do you prefer?
[data-test="loading"] (I started to using it) or [data-test-loading=""]?
If you are going to choose the first one, please change your attributes to its pattern. In other case please change my attributes to your pattern.

Reply:
I guess I was very sleepy :P.
You're right, we should stick to one convention, and it should be `data-test="loading"`

Reply:
ok